### PR TITLE
Angular: Add a component snippet format

### PR DESCRIPTION
**Pull request by valentinpalkovic**
## What I did

The snippet the earlier slices produce is a bare template fragment:

```html
<sb-button [label]="'Save'" [count]="3" (clicked)="clicked($event)"></sb-button>
```

That reads fine next to a rendered story, but you cannot do anything with it. An Angular template only exists inside a component, so there is nowhere to paste this, and an agent handed it has to guess the rest.

`framework.options.snippetFormat` picks the shape:

```ts
framework: {
  name: '@storybook/angular-vite',
  options: { snippetFormat: 'component' }, // 'template' is the default
}
```

`'component'` wraps the same markup in the host it needs to compile, and the payload grows the matching import block. Both payloads below are real output for the same fixture, captured from a run through the analyzer:

```jsonc
// snippetFormat: 'template'
{
  "id": "example-button",
  "name": "ButtonComponent",
  "path": "capture-tmp/button.stories.ts",
  "stories": {
    "example-button--basic": {
      "id": "example-button--basic",
      "name": "Basic",
      "snippet": "<sb-button [label]=\"'Save'\" [count]=\"3\" (clicked)=\"clicked($event)\"></sb-button>"
    }
  }
}
```

```jsonc
// snippetFormat: 'component'
{
  "id": "example-button",
  "name": "ButtonComponent",
  "path": "capture-tmp/button.stories.ts",
  "import": "import { Component } from '@angular/core';\nimport { ButtonComponent } from './button.component';",
  "stories": {
    "example-button--basic": {
      "id": "example-button--basic",
      "name": "Basic",
      "snippet": "@Component({\n  selector: 'app-root',\n  template: `<sb-button [label]=\"'Save'\" [count]=\"3\" (clicked)=\"clicked($event)\"></sb-button>`,\n  imports: [ButtonComponent],\n})\nexport class App {\n  clicked(event: unknown) {}\n}"
    }
  }
}
```

Which renders as:

```ts
import { Component } from '@angular/core';
import { ButtonComponent } from './button.component';

@Component({
  selector: 'app-root',
  template: `<sb-button [label]="'Save'" [count]="3" (clicked)="clicked($event)"></sb-button>`,
  imports: [ButtonComponent],
})
export class App {
  clicked(event: unknown) {}
}
```

Where the wrapper sits relative to what the earlier slices already do:

```
story file ──▶ CSF parse ──▶ markup ─────────────────────────────▶ story.snippet
                                │   (template: unchanged)
                                │
                                └──▶ scan markup for the output ──▶ angularHostComponent
                                     bindings it actually carries        │
                                                                         ▼
                                                            story.snippet ('component')
component meta ──▶ class name + export name + specifier ──▶ angularHostImports
                                                                         │
                                                                         ▼
                                                                  payload.import
```

## Three decisions worth a look

**Handlers are read back off the markup, not assumed.** Angular's template type checking resolves `(clicked)="clicked($event)"` against the host class, so a wrapper missing that method emits an example that does not compile, which would defeat the point of the format. Declaring a method for every output is wrong in the other direction: `argsToTemplate` is the idiom the Angular docs use everywhere, and it honours `include` / `exclude`, so a story that filtered an output out would get a member its template never references. The wrapper asks the markup instead:

```ts
const boundOutputs = (markup: string, outputs: readonly string[]): string[] =>
  outputs.filter((name) => markup.includes(bindingAttributes({ inputs: [], outputs: [name] })[0]));
```

Reusing `bindingAttributes` rather than a fresh regex means the check cannot drift from the grammar that produced the markup in the first place.

**A payload with no snippets gets no import block.** The provider deliberately still emits description-only stories when the analyzer cannot resolve a component. Advertising an import block for markup that was never produced would make those payloads look like they carry code. Both the wrapper and `payload.import` are gated on a snippet actually existing.

**`'template'` stays the default.** Flipping it is a one-line change to `DEFAULT_SNIPPET_FORMAT`, but leaving it means this PR changes nothing for existing users, and the 15 recorded `server-snippet-*.snapshot` browser-parity baselines stay valid untouched.

| Command | What it does | Run from |
| --- | --- | --- |
| `yarn test --project "@storybook/angular-vite"` | Both formats, plus wrapper edge cases: default and aliased exports, selector-less components, non-identifier output names, template-literal escaping | repo root |
| `node scripts/check/check-package.ts --cwd code/frameworks/angular-vite` | Type-checks the package on its own | repo root |

A wrapper that guessed its handlers instead of reading them fails like this:

```
 × src/docgen/story-docs-build.test.ts > buildStoryDocsPayload in component format > declares handlers only for the outputs the story kept
   → expected '<sb-button [label]="\'Save\'" (clicke…' to contain 'clicked(event: unknown) {}'
 × src/docgen/story-docs-build.test.ts > buildStoryDocsPayload in component format > exposes the class the outlet template reads as a value
   → expected '<ng-container *ngComponentOutlet="But…' to be '@Component({\n  selector: \'app-root\…' // Object.is equality
```

#### The changes in this PR are covered in the following automated tests:

- [ ] stories
- [x] unit tests
- [x] integration tests
- [ ] end-to-end tests

#### Manual testing

1. `yarn task sandbox --template angular-vite/docgen-server-ts --start-from auto`, then `yarn storybook` from the generated sandbox. That template is the Angular sandbox with `experimentalDocgenServer` and `componentsManifest` already on.
2. Open `Example/Button` → `Docs` and press **Show code**. You should see the bare `<sb-button …>` element.
3. Stop Storybook, set `framework.options.snippetFormat` to `'component'` in the sandbox's `.storybook/main.ts`, and start it again.
4. Press **Show code** on the same story. You should now see a full `@Component({ … }) export class App` block with a `clicked` method, instead of the bare element.
5. Remove the option again and confirm the block goes back to the template fragment.

### Documentation

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [ ] Declare whether manual QA will be needed for this PR during the next release, through `qa:needed` or `qa:skip`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

### CODE REVIEW

File: code/frameworks/angular-vite/src/docgen/component-snippet.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/**
+ * The standalone host component a story's markup is wrapped in, for the `component` snippet format.
+ *
+ * The `template` format emits the markup on its own. That reads well beside a rendered story but
+ * cannot be pasted anywhere, because an Angular template only exists inside a component. This
+ * format adds the host, the standalone imports it needs and the members its template references, so
+ * the snippet stands on its own.
+ */
+import { IDENTIFIER } from './template-snippet.ts';
```
Comment (valentinpalkovic) [inline code/frameworks/angular-vite/src/docgen/component-snippet.ts]:
```suggestion
import { IDENTIFIER } from './template-snippet.ts';
```
